op/model.py

```diff
@@ -237,6 +294,63 @@ def __init__(self, backend):
     def _load(self):
         return self._backend.config_get()
 
+class Status(ABC):
+    """Status values specific to applications and units."""
+
+    _statuses = {}
+
+    def __init__(self, message):
+        self.message = message
+
+    def __new__(cls, *args, **kwargs):
+        if cls is Status:
+            raise TypeError("cannot instantiate a base class")
+
+        return super().__new__(cls)
```

Comment:
What about args and kwargs?

Empty line above can go as well. These are common patterns:
```
if ...:
        return ...
return ...
```
and also
```
if ...:
        raise ...
return ...
```
or
```
if ...:
        return ...
raise ...
```

Reply:
It will fail with args and kwargs:

https://github.com/python/cpython/blob/3.7/Objects/typeobject.c#L3694-L3702

```
    return super().__new__(cls, *args, **kwargs)
TypeError: object.__new__() takes exactly one argument (the type to instantiate)
```